_example/simple/Dockerfile

```diff
@@ -0,0 +1,41 @@
+# =============================================================================
+#  Multi-stage Dockerfile Example
+# =============================================================================
+#  This is a simple Dockerfile that will build an image of scratch-base image.
+#  Usage:
+#    docker build -t simple:local . && docker run --rm simple:local
+# =============================================================================
+
+# -----------------------------------------------------------------------------
+#  Build Stage
+# -----------------------------------------------------------------------------
+FROM golang:alpine AS build
+
+RUN apk add --no-cache \
+    # Build essentials for development on Alpine
+    alpine-sdk build-base \
+    # go-sqlite3 requirement
+    sqlite-dev
+
+WORKDIR /workspace
+
+COPY . /workspace/
+
+RUN \
+    go mod init github.com/mattn/sample && \
+    go mod tidy && \
+    go install -ldflags="-s -w -extldflags \"-static\"" ./simple.go
+
+RUN \
+    # Smoke tests
+    which simple && \
```

Comment:
I don't think there's any need for the `which` command here. If the binary doesn't exist then invoking it will fail. On that note, I think it would be better to invoke it via the absolute path, to be sure you are calling the right thing.
Also, you should add `set -o pipefail` since by default `|` only fails if the rightmost command fails.

Reply:
You got the point. I now realize that I am making a "if-then-else" case!

How about this?

```dockerfile
RUN /go/bin/simple | grep 99\ こんにちわ世界099
```

Reply:
You should still add `set -o pipefail` or else only the exit code from `grep` will be considered.

Reply:
If `go/bin/simple` fails, then `grep` will fail because it cannot catch the "`99\ こんにちわ世界099`". Isn't it a bit redundant?

Reply:
Never mind. Let's be cautious!

Reply:
It depends where it fails. If it fails after emitting that message (perhaps because it crashes), then without `pipefail` it would get masked.